From d7364862d8f902c554a79e31ea285bb6e89ebd5b Mon Sep 17 00:00:00 2001
From: Anisetti Avinash Krishna <anisetti.avinash.krishna@intel.com>
Date: Fri, 29 May 2026 13:24:15 +0000
Subject: [PATCH] Allowlist updater privapp permissions

Add platform privapp allowlist entries for SystemUpdaterSample so
required privileged OTA permissions are granted at boot.

Tracked-On: NIACP3-1352
Signed-off-by: Anisetti Avinash Krishna <anisetti.avinash.krishna@intel.com>
---
 data/etc/privapp-permissions-platform.xml | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/data/etc/privapp-permissions-platform.xml b/data/etc/privapp-permissions-platform.xml
index 62e14d368a1c..fb7c5b548e94 100644
--- a/data/etc/privapp-permissions-platform.xml
+++ b/data/etc/privapp-permissions-platform.xml
@@ -42,6 +42,12 @@ applications that come with the platform
         <permission name="android.permission.SET_BIOMETRIC_DIALOG_ADVANCED" />
     </privapp-permissions>
 
+    <privapp-permissions package="com.example.android.systemupdatersample">
+	<permission name="android.permission.ACCESS_CACHE_FILESYSTEM"/>
+	<permission name="android.permission.RECOVERY"/>
+	<permission name="android.permission.UPDATE_PACKAGES"/>
+    </privapp-permissions>
+
     <privapp-permissions package="com.android.externalstorage">
         <permission name="android.permission.MOUNT_UNMOUNT_FILESYSTEMS"/>
         <permission name="android.permission.WRITE_MEDIA_STORAGE"/>
-- 
2.34.1

